### PR TITLE
[ENHC0010017] Change  download paths to use rails_blob_path

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR changes from using the download endpoints for sample attachments and data exports to utilizing the Rails helper `rails_blob_path` to download the files.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. Ensure you are able to create and download both sample and workflow execution exports
2. Ensure you can download both single and paired-end sample files.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/controllers/data_exports_controller.rb
Comment (ericenns) [inline app/controllers/data_exports_controller.rb:1]: Can you remove the reference to `download` in the before_action at the top of this file.

File: app/controllers/projects/samples/attachments_controller.rb
Comment (ericenns) [inline app/controllers/projects/samples/attachments_controller.rb:1]: Can you remove the reference to `download` in the before_action at the top of this file.

**Review by ericenns — CHANGES_REQUESTED** [review]
Just a few minor comments, otherwise this is working as expected.

File: app/controllers/data_exports_controller.rb
Comment (ChrisHuynh333) [reply app/controllers/data_exports_controller.rb:1]: Removed in 9284422be5d0dc2b62cf4205548c94afb5994127

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**